portfolio/src/main/webapp/js/comments.js

```diff
@@ -14,8 +31,44 @@ function initCommentsSystem() {
     event.preventDefault();
     loadComments();
   });
+}
+
+/**
+ * Updates the comments UI according to user authorization. 
+ */
+function applyAuthorizationToUI() {
+  if (commentAuthData.authorized) {
+    const withAuthElements = 
+        Array.from(document.getElementsByClassName("only-display-with-auth"));
+    // Make authorization-only UI visible
+    withAuthElements.forEach((elem) => elem.style.display = "inherit");
+  } else {
+    const withoutAuthElements = Array.from(
+        document.getElementsByClassName("only-display-without-auth"));
+    // Make no-authorization-only UI visible
+    withoutAuthElements.forEach((elem) => elem.style.display = "inherit");
+    
+    const loginButton = document.getElementById("comment-auth-login");
+    loginButton.href = commentAuthData.loginUrl;
+  }
+}
```

Comment:
As discussed above this function should take `commentAuthData` as an argument rather than reading it from a global variable.

```suggestion
function applyAuthorizationToUI(commentAuthData) {
  if (commentAuthData.authorized) {
    const withAuthElements = 
        Array.from(document.getElementsByClassName("only-display-with-auth"));
    // Make authorization-only UI visible
    withAuthElements.forEach((elem) => elem.style.display = "inherit");
  } else {
    const withoutAuthElements = Array.from(
        document.getElementsByClassName("only-display-without-auth"));
    // Make no-authorization-only UI visible
    withoutAuthElements.forEach((elem) => elem.style.display = "inherit");
    
    const loginButton = document.getElementById("comment-auth-login");
    loginButton.href = commentAuthData.loginUrl;
  }
}
```